app/assets/javascripts/discourse/app/mixins/quote-button.js

```diff
@@ -0,0 +1,296 @@
+import Mixin from "@ember/object/mixin";
+import { propertyEqual } from "discourse/lib/computed";
+import { selectedElement, selectedText } from "discourse/lib/utilities";
+import { INPUT_DELAY } from "discourse-common/config/environment";
+import KeyEnterEscape from "discourse/mixins/key-enter-escape";
+import { action } from "@ember/object";
+import discourseDebounce from "discourse-common/lib/debounce";
+import { schedule } from "@ember/runloop";
+import toMarkdown from "discourse/lib/to-markdown";
+import { regexSafeStr } from "discourse/lib/quote";
+
+export default Mixin.create(KeyEnterEscape, {
+  classNames: ["quote-button"],
+  classNameBindings: ["visible", "_displayFastEditInput:fast-editing"],
+  visible: false,
+  editText: null,
+
+  _isFastEditable: false,
+  _displayFastEditInput: false,
+  _fastEditInitalSelection: null,
+  _fastEditNewSelection: null,
+  _isSavingFastEdit: false,
+  _canEditPost: false,
+
+  _isMouseDown: false,
+  _reselected: false,
+
+  _hideButton() {
+    this.quoteState.clear();
+    this.set("visible", false);
+
+    this.set("_isFastEditable", false);
+    this.set("_displayFastEditInput", false);
+    this.set("_fastEditInitalSelection", null);
+    this.set("_fastEditNewSelection", null);
+  },
+
+  _selectionChanged() {
+    if (this._displayFastEditInput) {
+      return;
+    }
+
+    const quoteState = this.quoteState;
+
+    const selection = window.getSelection();
+    if (selection.isCollapsed) {
+      if (this.visible) {
+        this._hideButton();
+      }
+      return;
+    }
+
+    // TODO (martin) Allow for > 1 element's worth of content for chat,
+    // post only allows one to be selected.
+    let firstRange, requiredDataForQuote;
+    for (let r = 0; r < selection.rangeCount; r++) {
+      const range = selection.getRangeAt(r);
+      const $selectionStart = $(range.startContainer);
```

Comment:
Do you think it would be complicated to strip out the jQuery usage while we're at this?

Reply:
Not too bad, I was hoping to avoid it but I will do it now that we are delaying this to post-release :)